File: Navtrack.Web/Model/Models/DeviceModel.cs
using System.ComponentModel.DataAnnotations;

namespace Navtrack.Web.Model.Models
{
    public class DeviceModel
    {
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; }
        
        [Required]
        [MinLength(15)]
        public string IMEI { get; set; }

        public string Type { get; set; }
        
        public int DeviceTypeId { get; set; }
    }
}

File: Navtrack.Web/ClientApp/src/services/api/types/device/DeviceModel.ts
export type DeviceModel = {
  id: number,
  imei: string,
  name: string,
  type: string,
  deviceTypeId: number
};

export const DefaultDeviceModel : DeviceModel = {
  id: 0,
  imei: "",
  name: "",
  type: "",
  deviceTypeId: 0
};